gm4_live_catch/data/gm4_live_catch/functions/fish/cod/summon.mcfunction

```diff
@@ -0,0 +1,17 @@
+# run from gm4_live_catch:fish/FISH_TYPE/target
+
+
+# store nbt
+data modify storage gm4_live_catch:temp target set from entity @s
+
+# summon fish
+summon cod ~ ~1 ~ {Air:60,Health:1f,Tags:["gm4_lc_fish_new"]}
+
+# copy motion from item into fish
+data modify entity @e[type=cod,limit=1,sort=nearest,tag=gm4_lc_fish_new] Motion set from storage gm4_live_catch:temp target.Motion
```

Comment:
You're not using this storage, so you might as well skip the storage entirely -- at least with the current setup. This might change in case you actually need to read more data from NBT.
```suggestion
data modify entity @e[type=cod,limit=1,sort=nearest,tag=gm4_lc_fish_new] Motion set from entity @s Motion
```

This also applies to the other types of fish.